lib/Finance/Quote/FinanceAPI.pm

```diff
@@ -0,0 +1,276 @@
+our $METHODHASH = {subroutine => \&financeapi, 
+                   display => $DISPLAY, 
+                   labels => \@LABELS,
+                   features => $FEATURES};
+
+sub methodinfo {
+    return ( 
+        financeapi   => $METHODHASH,
+        nyse         => $METHODHASH,
+        nasdaq       => $METHODHASH,
+        usa          => $METHODHASH,
+    );
+}
+
+sub labels { my %m = methodinfo(); return map {$_ => [@{$m{$_}{labels}}] } keys %m; }
+
+sub methods {
+  my %m = methodinfo(); return map {$_ => $m{$_}{subroutine} } keys %m;
+}
+
+sub financeapi {
+
+  my $quoter = shift;
+  my @stocks = @_;
+  my (%info, $url, $reply);
+  my $ua = $quoter->user_agent();
+
+  # Set token. If not passed in as argument, get from environment.
+  my $token = exists $quoter->{module_specific_data}->{stockdata}->{API_KEY} ? 
+              $quoter->{module_specific_data}->{stockdata}->{API_KEY}        :
+              $ENV{"FINANCEAPI_API_KEY"};
+
+  # Set headers. API key is sent as a header.
+  my @ua_headers = (
+    'Accept' => 'application/json',
+    'X-API-KEY' => $token,
+  );
+
+  foreach my $stock (@stocks) {
+
+    if ( !defined $token ) {
+      $info{ $stock, 'success' } = 0;
+      $info{ $stock, 'errormsg' } =
+        'FINANCEAPI_API_KEY not defined. Get an API key at https://financeapi.net/';
+      next;
+    }
+
+    $url   = $FINANCEAPI_URL . $stock;
+    $reply = $ua->get( $url, @ua_headers );
+
+    my $code    = $reply->code;
+    my $desc    = HTTP::Status::status_message($code);
+    my $headers = $reply->headers_as_string;
+    my $body    = decode('UTF-8', $reply->content);
+
+    ### Body: $body
+
+    my ($quote, $name, $bid, $ask, $last, $open, $high, $low, $volume, $currency, $date);
+    my ($month, $day, $year);
+
+    $info{ $stock, "symbol" } = $stock;
+
+    if ( $code == 200 ) {
+
+      eval {$quote = JSON::decode_json $body};
+      if ($@) {
+        $info{ $stock, 'success' } = 0;
+        $info{ $stock, 'errormsg' } = $@;
+        next;
+      }
+
+      ### [<now>] JSON quote: $quote
+
+      if (!exists $quote->{'quoteResponse'}) {
+        $info{ $stock, 'success' } = 0;
+        $info{ $stock, 'errormsg' } = $@;
+        next;
+      }
+
+      $name     = $quote->{'quoteResponse'}{'result'}[0]{'longName'};
+      $open     = $quote->{'quoteResponse'}{'result'}[0]{'regularMarketOpen'};
+      $high     = $quote->{'quoteResponse'}{'result'}[0]{'regularMarketDayHigh'};
+      $low      = $quote->{'quoteResponse'}{'result'}[0]{'regularMarketDayLow'};
+      $last     = $quote->{'quoteResponse'}{'result'}[0]{'regularMarketPrice'};
+      $volume   = $quote->{'quoteResponse'}{'result'}[0]{'regularMarketVolume'};
+      #$currency = $quote->{'quoteResponse'}{'result'}[0]{'financialCurrency'};
+      $currency = $quote->{'quoteResponse'}{'result'}[0]{'currency'};
+      $date     = $quote->{'quoteResponse'}{'result'}[0]{'regularMarketTime'};
```

Comment:
I'd declare an intermediate var for `$quote->{'quoteResponse'}{'result'}[0]` and then use that for clarity. That is just about coding style ...